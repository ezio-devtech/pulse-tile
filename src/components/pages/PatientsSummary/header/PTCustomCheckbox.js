import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Col } from 'react-bootstrap';

export default class PTCustomCheckbox extends PureComponent {
	static propTypes = {
		title: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
		isChecked: PropTypes.bool.isRequired,
		onChange: PropTypes.func,
		disabled: PropTypes.bool,
	};

	toggleCheckbox = () => {
	  const { disabled = false, onChange } = this.props;
		return !disabled && onChange(name);
  };

	render() {
		const { title, name, isChecked, disabled = false } = this.props;

		return (
      <Col xs={6} sm={4}>
        <div className="wrap-fcustominp">
          <div className={classNames('fcustominp-state', { disabled })} onClick={this.toggleCheckbox} >
            <div className="fcustominp">
              <input type="checkbox" id={`dashboard-${name}`} name={`dashboard-${name}`} checked={isChecked} onChange={this.toggleCheckbox} />
              <label htmlFor={`dashboard-${name}`} />
            </div>
            <label htmlFor={`dashboard-${name}`} className="fcustominp-label">{title}</label>
          </div>
        </div>
      </Col>
    )
  }
};
