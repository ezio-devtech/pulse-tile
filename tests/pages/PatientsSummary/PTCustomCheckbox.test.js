import React from 'react';
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });

import PTCustomCheckbox from '../../../src/components/pages/PatientsSummary/header/PTCustomCheckbox';

const testProps = {
	onChange: () => {},
	title: 'Test title',
	name: 'test-name',
};

describe('Component <PTCustomCheckbox />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
        <PTCustomCheckbox
          onChange={testProps.onChange}
          title={testProps.title}
          name={testProps.name}
          isChecked={false}
        />);
		expect(component).toMatchSnapshot();

		expect(component.instance().props['onChange']).toEqual(testProps.onChange);
		expect(component.instance().props['title']).toEqual(testProps.title);
		expect(component.instance().props['name']).toEqual(testProps.name);
		expect(component.instance().props['isChecked']).toEqual(false);
		expect(component.instance().props['disabled']).toEqual(undefined);

    expect(component.find('input')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(2);

    const idInput = `dashboard-${testProps.name}`;
		expect(component.find('input').prop('type')).toEqual('checkbox');
		expect(component.find('input').prop('id')).toEqual(idInput);
		expect(component.find('input').prop('name')).toEqual(idInput);
		expect(component.find('label').at(0).prop('htmlFor')).toEqual(idInput);
		expect(component.find('label').at(1).prop('htmlFor')).toEqual(idInput);

		component.instance().toggleCheckbox();
		component.setProps({ disabled: true});
		expect(component.instance().props['disabled']).toEqual(true);
		component.instance().toggleCheckbox();
  });
});
