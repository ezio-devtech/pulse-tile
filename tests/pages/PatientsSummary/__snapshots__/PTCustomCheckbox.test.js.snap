// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PTCustomCheckbox /> should renders with all props correctly 1`] = `
<Col
  bsClass="col"
  componentClass="div"
  sm={4}
  xs={6}
>
  <div
    className="wrap-fcustominp"
  >
    <div
      className="fcustominp-state"
      onClick={[Function]}
    >
      <div
        className="fcustominp"
      >
        <input
          checked={false}
          id="dashboard-test-name"
          name="dashboard-test-name"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="dashboard-test-name"
        />
      </div>
      <label
        className="fcustominp-label"
        htmlFor="dashboard-test-name"
      >
        Test title
      </label>
    </div>
  </div>
</Col>
`;
